segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/DebugBookKeeperLogWrapper.java

```diff
@@ -151,16 +151,19 @@ public void disable() throws DurableDataLogException {
      * @param candidateLedgers A List of {@link LedgerHandle}s that contain all the Ledgers that this {@link BookKeeperLog}
      *                         should contain. This could be the list of all BookKeeper Ledgers or a subset, as long as
      *                         it contains all Ledgers that list this {@link BookKeeperLog} as their owner.
+     * @param overwrite  Boolean denoting whether reconcile has to be forced even if log is enabled
      * @return True if something changed (and the metadata is updated), false otherwise.
      * @throws IllegalStateException   If this BookKeeperLog is not disabled.
      * @throws DurableDataLogException If an exception occurred while updating the metadata.
      */
-    public boolean reconcileLedgers(List<? extends ReadHandle> candidateLedgers) throws DurableDataLogException {
+    public boolean reconcileLedgers(List<? extends ReadHandle> candidateLedgers, boolean overwrite) throws DurableDataLogException {
```

Comment:
Why to change this method? Is it not valid to require the Bookkeeper Log to be disabled in order to perform the reconciliation? Actually, in the case of a missing ledger, the Segment Container would be setting it as "disabled" automatically after getting a `DataCorruptionException`. Please, provide a good reason for changing it and if we can live with the previous version, please revert this change.